crates/compiler/gen_dev/src/generic64/mod.rs

```diff
@@ -1544,7 +1544,7 @@ impl<
         match self.interner().get(*arg_layout).repr {
             single_register_int_builtins!() | LayoutRepr::BOOL => {
                 let width = match *arg_layout {
-                    Layout::BOOL | Layout::I8 | Layout::U8 => RegisterWidth::W8,
+                    Layout::BOOL_NO_SEMA | Layout::I8 | Layout::U8 => RegisterWidth::W8,
```

Comment:
We probably want this to really be

```
match self.interner().get(*arg_layout).repr {
  LayoutRepr::BOOL | LayoutRepr::I8 | LayoutRepr::U8 => ...
  ...
}
```

instead of loooking at the Layout, which can be semantic